### PR TITLE
HalfOfMonth variable not being used

### PR DESCRIPTION
**Pull request by Dannyps**
# Readiness checklist

-   [x] I added/updated unit tests.
-   [ ] I added/updated integration tests.
-   [x] I ensured that the PR title is good enough for the changelog.
-   [x] I labeled the PR.

# Description
When the `Counts_entities_within_timeframe_month_quotaPeriod` test was developed, we attempted to pivot all dates around the 15th of the current month so that we could use notions such as _Tomorrow_ or _Yesterday_ and be certain that those dates would fall on the same Month. Turns out that this was not done properly, which led to errors when running this test on the last/first day of each month.

trivia: The commit is dated from October 31st, but it was actually done about the same time as this PR. Changing the system date for debugging had such a side-effect, i.e., I committed my changes before setting the clock to the right date again.

### CODE REVIEW

File: Modules/Quotas/test/Quotas.Application.Tests/Tests/Repositories/MessagesRepositoryTests.cs [hunk 1]
```diff
@@ -14,16 +14,18 @@
 namespace Backbone.Modules.Quotas.Application.Tests.Tests.Repositories;
 public class MessagesRepositoryTests
 {
+    private static readonly DateTime HALF_OF_MONTH = new(DateTime.Now.Year, DateTime.Now.Month, 15);
```
Comment (tnotheis) [inline Modules/Quotas/test/Quotas.Application.Tests/Tests/Repositories/MessagesRepositoryTests.cs:17]: Can't we just call this `TODAY`? And then I would also set `SystemTime` to this value in the constructor of the test.

Otherwise, nobody understands why YESTERDAY is set to HALF_OF_MONTH-1

**Review by tnotheis — CHANGES_REQUESTED** [review]

File: Modules/Quotas/test/Quotas.Application.Tests/Tests/Repositories/MessagesRepositoryTests.cs (hunk 1 above)
Comment (Dannyps) [reply Modules/Quotas/test/Quotas.Application.Tests/Tests/Repositories/MessagesRepositoryTests.cs:17]: of course, that helps!

**Review by tnotheis — APPROVED** [review]

**Review by tnotheis — CHANGES_REQUESTED** [review]
Wait

**Comment by tnotheis** [comment]
Don't we have to set the SystemTime in the constructor?

**Comment by Dannyps** [comment]
We're setting the time in the `CreateMessage` method because it is only required for the arrangement of the tests. The Act and Assert can run with any time.

**Comment by Dannyps** [comment]
1. It would indeed make more sense to use TODAY, despite the tests being written in a way that works with the actual date. I'll make it so.
2. I see your point and even agree with it to some extent, but it should be noted that it was the fact that we did not hardcode a date that allowed us to find the bug we're now fixing, and I wonder what else may come out of that.
3. Of course.

**Comment by tnotheis** [comment]
> it should be noted that it was the fact that we did not hardcode a date that allowed us to find the bug we're now fixing, and I wonder what else may come out of that.

But there is nothing worse than non-reproducible tests. 

If there is a bug your current tests don't "see", then you are missing a test. So maybe you should think about what edge cases could exist, and write tests for them.

**Comment by Dannyps** [comment]
You're correct. Sadly, I'm pretty confident that there are no further edge cases.

**Review by tnotheis — APPROVED** [review]
